src/v/datalake/tests/record_generator.cc

```diff
@@ -41,11 +49,113 @@ record_generator::register_avro_schema(
     co_return std::nullopt;
 }
 
+ss::future<checked<std::nullopt_t, record_generator::error>>
+record_generator::register_protobuf_schema(
+  std::string_view name, std::string_view schema) {
+    using namespace pandaproxy::schema_registry;
+    auto id = co_await ss::coroutine::as_future(
+      _sr->create_schema(unparsed_schema{
+        subject{"foo"},
+        unparsed_schema_definition{schema, schema_type::protobuf}}));
+    if (id.failed()) {
+        co_return error{fmt::format(
+          "Error creating schema {}: {}", name, id.get_exception())};
+    }
+    auto [_, added] = _id_by_name.emplace(name, id.get());
+    if (!added) {
+        co_return error{fmt::format("Failed to add schema {} to map", name)};
+    }
+    co_return std::nullopt;
+}
+
+iobuf encode_protobuf_message_index(const std::vector<int32_t>& message_index) {
```

Comment:
Is there any existing serializer for this message index format anywhere in our code-base? There is `get_proto_offsets` in `src/v/datalake/schema_registry.h`. Happy to move this serializer to a more general location if there is any use for it outside of the record generator.

Reply:
Closest thing is https://github.com/redpanda-data/redpanda/blob/514f1df4088dcd639a9bd6f1a153460c3f1f35a8/src/v/datalake/tests/record_schema_resolver_test.cc#L63-L72

I don't have strong feelings about code placement, I think leaving it in the record generate seems reasonable